cmd/builder/internal/builder/main_test.go

```diff
@@ -258,6 +258,8 @@ func TestGenerateAndCompile(t *testing.T) {
 			testCase: "Default Configuration Compilation",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
+				err := cfg.SetBackwardsCompatibility()
```

Comment:
These tests all pass in a list of providers, and therefore depend on those providers in the go mod list.  The builder templates only add them when `.Distribution.SupportsConfmapFactories` is true.  That value was not being set so all these tests should have been failing.  They were saved instead by the fact that otelcol depended on the providers.  The changes in this PR remove those dependencies when `otelcoltest` is not being used, so these tests started failing.  The fix was to make sure `.Distribution.SupportsConfmapFactories` is true, which is included in `cfg.SetBackwardsCompatibility()`.

Reply:
Nice catch, thanks for the explanation.